corehq/warehouse/models/dimensions.py

```diff
@@ -0,0 +1,86 @@
+from django.db import models
+
+
+class BaseDim(models.Model):
+    domain = models.CharField(max_length=255)
+
+    dim_last_modified = models.DateTimeField(auto_now=True)
+    dim_created_on = models.DateTimeField(auto_now_add=True)
+
+    class Meta:
+        abstract = True
+
+
+class UserDim(BaseDim):
+    user_id = models.CharField(max_length=255)
+    username = models.CharField(max_length=255)
```

Comment:
I think username, first and last name can all be reduced. https://docs.djangoproject.com/en/1.11/ref/contrib/auth/ (I think first and last may be getting an increase in 2.0)

Unsure about email though. maybe it's a text field in django? Either way we should keep in consistent